### ISSUE TITLE
Make breadcrumbs semantic aware, flexible and fast

### ISSUE THREAD

**Issue by DiegoPino** [esmero/strawberryfield#68]
# What is that i want?

Everybody knows and loves breadcrumbs. But hey. In semantic, linked data world they are like looking trough a key lock the whole reality. Simple use case everyone knows is Collection/member. But that is not all, E.g, an ADO can be part of two collections, or can be connected to other objects via `isrelatedto`, `partof`, `sameas`, etc, etc. All those things are relevant when thinking about a breadcrumb that is useful. People should be able to check which ones they want in the breadcrumb

So what i want?

Breadcrumbs that read from a list of JSON keys (you define which keys, we could even tag our JMESPATHs that are referencing NODES as such), then fast traverse (i'm quite good at traversing graphs) and accumulate not a single hierarchy but a list of parents. And so on. Direction of the relationship is important. We can do this via direct entityQueries (slow but consistent) or we can use Solr to drive this, fast but requires setting each predicate that people want as Fields, so more config.

UI is tricky. How to draw this tree (its not a cylic graph) in a way that still makes sense to people? Maybe show only one path by default and expand, via JS to more (if there is more) on over any node in this tree? Maybe color coded or prefixed with an Unicode character? having this would be lovely.

**Comment by marlo-longley** [comment]
@DiegoPino I have been thinking about this post!

I really like the functionality and relationships that you describe here. isrelatedto, partof, sameas, multiple parent collections, should all be available information for ADOs and will make Archipelago really rich with info. I am wondering if, trying to force this into the concept of breadcrumbs is too much. I feel that breadcrumbs are known and loved because they are simple and linear, and orient a user to their location in a sitemap. I would love to see a sketch of what you are thinking, because it sounds very cool! I researched existing tree UIs, most are similar to a filesystem/folder browsing, or like an org/flow chart.

On the ADO page itself, to start, all the relationships and collections it is a member of could be listed and hyperlinked in an orderly/pleasing way. Under "Collections" there could be 2 items listed, etc. People can define which relationships they want to show in ADO views like you say. Then we could refine the UI in later steps?

Most people will arrive at an ADO through searching or browsing, and it is the page/view directly prior to reaching the ADO itself where I can more imagine more navigation elements. What do you think?

**Comment by DiegoPino** [comment]
@marlo-longley i agree with the fact that breadcrumbs should be simple/flat. But i also feel, as you say, we need flexible ways of showing the rich complexities of relatationships between ADOs.

This is what i would like:
A setting page where people can select, using the existing Terms (or on an empty archipelagos) which JSON keys are bearers of relationships between entities. And give them a Weight, like a Table drag. Normal Breadcrump would then use these to build a flat breadcrumb using the first (based on that weight) relationship it finds and build the hierarchy. Then another Block/formatter could use all relationships to build a full graph that can be but somewhere (i already have the super simple logic for that).

Does that sound OK?

About
> In the ADO page itself, to start, all the relationships and collections it is a member of could be listed and hyperlinked in an orderly/pleasing way. Under "Collections" there could be 2 items listed, etc. People can define which relationships they want to show in ADO views like you say. Then we could refine the UI in later steps?

Yes: That can/is solved (but needs to be prettier) using a simple View. IF we can feed the relationships people define into the view dynamically we could even get the best of two worlds.

Also, lasty, we should make a V2 of this that only uses SOLR. To speed things up.

**Comment by marlo-longley** [comment]
@DiegoPino thanks, let me think more about this, but I also thought this was the type of convo @mitchellkeaney might be interested in at least reading so tagging him.

**Comment by giancarlobi** [comment]
@DiegoPino @marlo-longley @mitchellkeaney just another point of view about this: standard breadcrumb make sense with standard website but Archipelago is not a standard website while more like a digital library with digital objects. Metadata are assigned to the objects by SBF and metadata are very very important for digital asset curators and metadata are the source of relationships between objects. So breadcrumbs based on SBF Json make a lot of sense in this case and you can think breadcrumb like a simple LOD graph to browse repository.
